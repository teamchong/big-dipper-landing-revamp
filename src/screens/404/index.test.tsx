import React from 'react';
import { render, screen, act } from '@testing-library/react';
import { MockTheme } from '@mocks';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@src/components/layout', () => (props: object) => <div data-testid="Layout" {...props} />);
jest.mock('@src/components/section-box', () => ({ main, ...props }: { [p: string]: unknown }) => (
  <div data-testid="SectionBox" {...props} />
));
// ==================================
// unit tests
// ==================================
describe('404', () => {
  it('matches snapshot', async () => {
    await act(async () => {
      global.innerWidth = 1400;
      global.dispatchEvent(new Event('resize'));
    });

    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>,
    );

    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByText('404')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
