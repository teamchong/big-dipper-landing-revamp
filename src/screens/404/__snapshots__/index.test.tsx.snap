// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 matches snapshot 1`] = `
.emotion-0>div>div:nth-of-type(2) {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column nowrap;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-0 img {
  margin-top: -20px!important;
  margin-bottom: -100px!important;
}

.emotion-0 h3 {
  font-size: 1.2rem;
}

.emotion-0 h4 {
  font-size: 0.95rem;
}

@media (min-width:768px) {
  .emotion-0 h3 {
    font-size: 2rem;
  }

  .emotion-0 h4 {
    font-size: 1.4rem;
  }
}

.emotion-1 {
  margin: 0;
  font-size: 1.5rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.167;
}

.emotion-2 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.235;
}

<div>
  <div
    class="emotion-0"
    data-testid="Layout"
  >
    <img
      alt="404"
      height="380"
      src="/images/404-unicorn.svg"
      width="380"
    />
    <h3
      class="MuiTypography-root MuiTypography-h3 emotion-1"
    >
      404
    </h3>
    <p>
       
    </p>
    <h4
      class="MuiTypography-root MuiTypography-h4 emotion-2"
    >
      404Description
    </h4>
  </div>
</div>
`;
